### ISSUE TITLE
Compilation Error

### ISSUE THREAD

**Issue by ShawnMcCool** [slashdotdash/commanded-extreme-adapter#5]
```
Resolving Hex dependencies...
Dependency resolution completed:
  certifi 1.2.1
  commanded 0.14.0
  commanded_ecto_projections 0.6.0
  commanded_extreme_adapter 0.2.0
  connection 1.0.4
  cowboy 1.1.2
  cowlib 1.0.2
  db_connection 1.1.2
  decimal 1.4.0
  ecto 2.2.6
  exprotobuf 1.2.9
  extreme 0.9.1
  file_system 0.2.1
  gettext 0.13.1
  gpb 3.27.7
  hackney 1.8.6
  httpoison 0.11.2
  idna 5.0.2
  mariaex 0.8.3
  metrics 1.0.1
  mime 1.1.0
  mimerl 1.0.2
  nadia 0.4.2
  phoenix 1.3.0
  phoenix_ecto 3.3.0
  phoenix_html 2.10.4
  phoenix_live_reload 1.1.2
  phoenix_pubsub 1.0.2
  plug 1.4.3
  poison 3.1.0
  poolboy 1.5.1
  postgrex 0.13.3
  ranch 1.3.2
  ssl_verify_fun 1.1.1
  table_rex 0.10.0
  unicode_util_compat 0.2.0
  uuid 1.1.8
➜  mccool-bot git:(master) ✗ mix deps.compile commanded_extreme_adapter
==> commanded_extreme_adapter
Compiling 4 files (.ex)

== Compilation error in file lib/extreme.ex ==
** (UndefinedFunctionError) function Commanded.EventStore.TypeProvider.__using__/1 is undefined or private
    Commanded.EventStore.TypeProvider.__using__([])
    lib/extreme.ex:14: (module)
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
could not compile dependency :commanded_extreme_adapter, "mix compile" failed. You can recompile this dependency with "mix deps.compile commanded_extreme_adapter", update it with "mix deps.update commanded_extreme_adapter" or clean it with "mix deps.clean commanded_extreme_adapter"
```

Hi there, can you help me understand why I'm having this compilation error?

**Comment by ShawnMcCool** [comment]
Is it potentially because of:

```
warning: the dependency :extreme requires Elixir "~> 1.3.0 or ~> 1.4.0" but you are running on v1.5.2
```

**Comment by slashdotdash** [comment]
@ShawnMcCool This is a bug caused by the latest Commanded v0.14.0 release which no longer exposes the macro used by this adapter. It needs to lookup the config at runtime, not compile time (using the macro).

I will fix the issue and publish an update to Hex.

**Comment by slashdotdash** [comment]
Published to Hex as [v0.3.0](https://hex.pm/packages/commanded_extreme_adapter/0.3.0).
